### PR TITLE
[shelly] Misc changes (small fixes, log improvements, hardened leak prevention on

### PR DESCRIPTION
**Pull request by markus7017**
This PR brings some smaller likes and improvements + hardened handling for leak prevention to avoid loosing sockets on initialization/discovery problems.

Closing #15919 
Closing #15920

### CODE REVIEW

**Comment by markus7017** [comment]
@lolodomo I put this small PR together to include small fixes before adding new features
We could consider to merge the last PR and this one also to 4.0,
otherwise I would at least add support for Shelly Dimmer 10V here to include some more meat
what do you think?

**Comment by markus7017** [comment]
@wborn Could you confirm hat 4.1 has jetty web socket server artifact as part of the core bundle so I could remove the dependency in pom. xml?

File: bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/api1/Shelly1CoapHandler.java [hunk 1]
```diff
@@ -265,9 +266,18 @@ public void processResponse(@Nullable Response response) {
...
+            // (e.g. auth failed, but device sends COAP packets via multicast)
+            if (thingHandler.getThingStatusDetail() == ThingStatusDetail.CONFIGURATION_ERROR) {
+                logger.debug("{}: The device is not configuired correctly, skip Coap packet", thingName);
+                return;
+            }
+
             // If we received a CoAP message successful the thing must be online
             thingHandler.setThingOnline();
 
+            thingHandler.setThingOnline();
```
Comment (lolodomo) [inline bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/api1/Shelly1CoapHandler.java:279]: Why calling twice consecutively ?

**Review by lolodomo — CHANGES_REQUESTED** [review]
Review part 1 of 2 (until Shelly2RpcSocket)

File: bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/api1/Shelly1CoapHandler.java (hunk 1 above)
Comment (markus7017) [reply bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/api1/Shelly1CoapHandler.java]: nope, somehow rebase created some duplicates
removed

**Comment by markus7017** [comment]
@lolodomo changes applied

**Comment by wborn** [comment]
> Could you confirm hat 4.1 has jetty web socket server artifact as part of the core bundle so I could remove the dependency in pom. xml?

Yes it would be better to remove it so it uses the core dependency at compile time:

https://github.com/openhab/openhab-core/blob/7af02598ef59b4998b19b31b49023b7f54e9cc39/bom/compile/pom.xml#L173-L178

**Comment by markus7017** [comment]
> > Could you confirm hat 4.1 has jetty web socket server artifact as part of the core bundle so I could remove the dependency in pom. xml?
> 
> Yes it would be better to remove it so it uses the core dependency at compile time:
> 
> https://github.com/openhab/openhab-core/blob/7af02598ef59b4998b19b31b49023b7f54e9cc39/bom/compile/pom.xml#L173-L178

ok, that's what #15920 brings in, just want to double check avoiding unresolved dependency on production systems

**Comment by markus7017** [comment]
> Yes it would be better to remove it so it uses the core dependency at compile time:

@wborn Does that also applies to 4.0.5+, so could this PR also go back to 4.0.x without causing trouble with unresolved dependencies?

@lolodomo What do think on merging this and the last PR to 4.0?

**Comment by wborn** [comment]
> Does that also applies to 4.0.5+, so could this PR also go back to 4.0.x without causing trouble with unresolved dependencies?

I think that should also work.

**Comment by markus7017** [comment]
@lolodomo fyi: we did additional testing with this code level and LGTM

File: bundles/org.openhab.binding.shelly/src/main/resources/OH-INF/i18n/shelly.properties [hunk 2]
```diff
@@ -247,7 +247,7 @@ channel-type.shelly.temperature4.description = Temperature of external Sensor #4
 channel-type.shelly.temperature5.label = Temperature 5
 channel-type.shelly.temperature6.description = Temperature of external Sensor #5
 channel-type.shelly.targetTemp.label = Target Temperature
-channel-type.shelly.targetTemp.description = Target Temperature  in �C to be reached in auto-temperature mode
+channel-type.shelly.targetTemp.description = Target Temperature in C to be reached in auto-temperature mode
```
Comment (lolodomo) [inline bundles/org.openhab.binding.shelly/src/main/resources/OH-INF/i18n/shelly.properties:250]: Why not keeping °C ?

**Review by lolodomo — CHANGES_REQUESTED** [review]
Final review. Just one small comment

File: bundles/org.openhab.binding.shelly/src/main/resources/OH-INF/i18n/shelly.properties (hunk 2 above)
Comment (markus7017) [reply bundles/org.openhab.binding.shelly/src/main/resources/OH-INF/i18n/shelly.properties]: is back

Comment (lolodomo) [reply bundles/org.openhab.binding.shelly/src/main/resources/OH-INF/i18n/shelly.properties]: Looks like it is not the right character.
image

**Comment by markus7017** [comment]
@lolodomo changes applied

**Review by lolodomo — APPROVED** [review]
LGTM, thank you

**Pull request merged**